### PR TITLE
change copy link logic and complete task

### PR DESCRIPTION
**Pull request by MishkaV**

### CODE REVIEW

File: app/src/main/kotlin/io/eugenethedev/taigamobile/ui/screens/main/MainActivity.kt [hunk 1]
```diff
@@ -232,11 +235,22 @@ fun MainScreen(
         }
     }
 
+    val scope = rememberCoroutineScope()
+    val context = LocalContext.current
+    val onShowMessage: (Int) -> Unit = { message ->
```
Comment (EugeneTheDev) [inline app/src/main/kotlin/io/eugenethedev/taigamobile/ui/screens/main/MainActivity.kt:240]: `showMessage` would look better

File: app/src/main/kotlin/io/eugenethedev/taigamobile/domain/entities/Task.kt [hunk 2]
```diff
@@ -73,6 +73,7 @@ data class CommonTaskExtended(
     val dueDate: LocalDate?,
     val dueDateStatus: DueDateStatus?,
     val userStoryShortInfo: UserStoryShortInfo? = null,
+    val uri: String,
```
Comment (EugeneTheDev) [inline app/src/main/kotlin/io/eugenethedev/taigamobile/domain/entities/Task.kt:76]: It should be `url`, not `uri`

File: app/src/main/kotlin/io/eugenethedev/taigamobile/ui/screens/commontask/components/CommonTaskAppBar.kt [hunk 3]
```diff
@@ -26,9 +28,11 @@ fun CommonTaskAppBar(
     commonTaskType: CommonTaskType,
     showTaskEditor: () -> Unit,
     editActions: EditActions,
-    navigationActions: NavigationActions
+    navigationActions: NavigationActions,
+    uri: String
```
Comment (EugeneTheDev) [inline app/src/main/kotlin/io/eugenethedev/taigamobile/ui/screens/commontask/components/CommonTaskAppBar.kt:32]: `url`

File: app/src/main/kotlin/io/eugenethedev/taigamobile/ui/screens/main/MainActivity.kt [hunk 4]
```diff
@@ -144,7 +146,8 @@ class MainActivity : AppCompatActivity() {
                                 ) {
                                     items.forEach { screen ->
                                         NavigationBarItem(
-                                            modifier = Modifier.navigationBarsPadding()
+                                            modifier = Modifier
```
Comment (EugeneTheDev) [inline app/src/main/kotlin/io/eugenethedev/taigamobile/ui/screens/main/MainActivity.kt:149]: Careful, autoformatting

File: app/src/main/kotlin/io/eugenethedev/taigamobile/ui/screens/main/MainActivity.kt [hunk 5]
```diff
@@ -232,11 +235,22 @@ fun MainScreen(
...
+            scaffoldState.snackbarHostState.showSnackbar(strMessage)
+        }
+    }
+
     val isLogged by viewModel.isLogged.collectAsState()
     val isProjectSelected by viewModel.isProjectSelected.collectAsState()
 
     Surface(
-        modifier = Modifier.fillMaxSize().padding(paddingValues),
+        modifier = Modifier
```
Comment (EugeneTheDev) [inline app/src/main/kotlin/io/eugenethedev/taigamobile/ui/screens/main/MainActivity.kt:251]: Autoformatting

**Review by EugeneTheDev — CHANGES_REQUESTED** [review]

File: app/src/main/kotlin/io/eugenethedev/taigamobile/ui/screens/main/MainActivity.kt (hunk 1 above)
Comment (MishkaV) [reply app/src/main/kotlin/io/eugenethedev/taigamobile/ui/screens/main/MainActivity.kt:240]: Did you mean to change `onShowMessage` to `showMessage`?

**Comment by EugeneTheDev** [comment]
Please pay attention to the commit above. If you want to invoke callback from `@Composable` function, always wrap it in the `LaunchedEffect`, `SideEffect` or `DisposableEffect`. Direct invocation is considered bad practice due to the poor perfomance and possible undesired behavior (because this callback will be invoked on each recomposition).

Other than that, LGTM

**Review by EugeneTheDev — APPROVED** [review]

**Pull request merged**